game/players.py

```diff
@@ -21,25 +21,32 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.distributions import Categorical
+import torch.optim as optim
 import numpy as np
 import random
-
+from types import SimpleNamespace
```

Comment:
Wasn't this approach scrapped? As far as I can see you moved away from it in your code, as we discussed yesterday.  
If I haven't missed something, please delete the import as to avoid confusion